File: app/src/main/java/com/marvelcomics/brito/marvelcomics/ui/fragment/series/SeriesAdapter.java
package com.marvelcomics.brito.marvelcomics.ui.fragment.series;

import android.content.Context;
import android.databinding.BindingAdapter;
import android.databinding.DataBindingUtil;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.marvelcomics.brito.entity.SeriesEntity;
import com.marvelcomics.brito.infrastructure.utils.MarvelThumbnailAspectRatio;
import com.marvelcomics.brito.marvelcomics.R;
import com.marvelcomics.brito.marvelcomics.databinding.ItemSeriesBinding;
import com.squareup.picasso.Picasso;

import java.util.ArrayList;
import java.util.List;

public class SeriesAdapter extends RecyclerView.Adapter<SeriesAdapter.ViewHolder> {

    private List<SeriesEntity> series = new ArrayList<>();

    public SeriesAdapter(List<SeriesEntity> series) {
        this.series = series;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        ItemSeriesBinding binding = DataBindingUtil.inflate(inflater, R.layout.item_series, parent, false);
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.bind(series.get(position));
    }

    @BindingAdapter("android:src")
    public static void setComicImage(ImageView imageView, SeriesEntity series) {
        Context context = imageView.getContext();
        Picasso.with(imageView.getContext())
                .load(series.getFullUrlThumbnailWithAspect(MarvelThumbnailAspectRatio.Portrait.XLARGE))
                .placeholder(context.getDrawable(R.drawable.progress))
                .fit()
                .centerInside()
                .into(imageView);
    }

    @Override
    public int getItemCount() {
        return series.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        private ItemSeriesBinding binding;

        ViewHolder(ItemSeriesBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        void bind(SeriesEntity seriesEntity) {
            binding.setSeries(seriesEntity);
            binding.executePendingBindings();
        }
    }
}

File: app/src/main/java/com/marvelcomics/brito/view/fragment/series/SeriesAdapter.kt
package com.marvelcomics.brito.view.fragment.series

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.marvelcomics.brito.R
import com.marvelcomics.brito.circularProgressBar
import com.marvelcomics.brito.data.entity.SeriesEntity
import com.marvelcomics.brito.infrastructure.utils.MarvelThumbnailAspectRatio

class SeriesAdapter(private val series: List<SeriesEntity>) :
    RecyclerView.Adapter<SeriesAdapter.ViewHolder?>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val view = layoutInflater.inflate(R.layout.item_series, null)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(series[position])
    }

    override fun getItemCount(): Int {
        return series.size
    }

    class ViewHolder(private val view: View) : RecyclerView.ViewHolder(view) {

        private var title: TextView? = null
        private var imageView: ImageView? = null

        init {
            title = view.findViewById(R.id.textview_title_series)
            imageView = view.findViewById(R.id.imageview_series)
        }

        fun bind(seriesEntity: SeriesEntity) {
            title?.text = seriesEntity.title
            imageView?.let {
                Glide.with(it.context)
                    .load(seriesEntity.getFullUrlThumbnailWithAspect(MarvelThumbnailAspectRatio.Portrait.SMALL))
                    .placeholder(view.context.circularProgressBar())
                    .fitCenter()
                    .into(it)
            }
        }
    }
}